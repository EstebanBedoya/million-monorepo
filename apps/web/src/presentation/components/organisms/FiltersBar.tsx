'use client';

import { useState, useEffect, useCallback, useRef } from 'react';
import { SearchInput } from '../molecules/SearchInput';
import { PropertyTypeSelect } from '../molecules/PropertyTypeSelect';
import { PriceRange } from '../molecules/PriceRange';
import { ClearFiltersButton } from '../molecules/ClearFiltersButton';

export interface FilterValues {
  search: string;
  minPrice: number;
  maxPrice: number;
  propertyType?: string;
}

interface FiltersBarProps {
  onFilterChange: (filters: FilterValues) => void;
  defaultFilters?: Partial<FilterValues>;
}

function useDebounce<T>(value: T, delay: number): T {
  const [debouncedValue, setDebouncedValue] = useState<T>(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
}

export function FiltersBar({ onFilterChange, defaultFilters }: FiltersBarProps) {
  const [search, setSearch] = useState(defaultFilters?.search || '');
  const [minPrice, setMinPrice] = useState(defaultFilters?.minPrice || 0);
  const [maxPrice, setMaxPrice] = useState(defaultFilters?.maxPrice || 1000000000);
  const [propertyType, setPropertyType] = useState(defaultFilters?.propertyType || '');
  
  const isFirstRender = useRef(true);
  const onFilterChangeRef = useRef(onFilterChange);

  const debouncedSearch = useDebounce(search, 400);

  // Update ref when callback changes
  useEffect(() => {
    onFilterChangeRef.current = onFilterChange;
  }, [onFilterChange]);

  useEffect(() => {
    // Skip first render to avoid calling on mount
    if (isFirstRender.current) {
      isFirstRender.current = false;
      return;
    }

    onFilterChangeRef.current({
      search: debouncedSearch,
      minPrice,
      maxPrice,
      propertyType,
    });
  }, [debouncedSearch, minPrice, maxPrice, propertyType]);

  const handleClearFilters = useCallback(() => {
    setSearch('');
    setMinPrice(0);
    setMaxPrice(1000000000);
    setPropertyType('');
  }, []);

  const hasActiveFilters = search || minPrice > 0 || maxPrice < 1000000000 || propertyType;

  return (
    <div className="space-y-6">
      <div className="space-y-4">
        <SearchInput
          id="search"
          label="Search by name or address"
          value={search}
          onChange={setSearch}
          placeholder="e.g., Modern Villa, Los Angeles..."
        />

        <PropertyTypeSelect
          id="property-type"
          label="Property Type"
          value={propertyType}
          onChange={setPropertyType}
        />
      </div>

      <PriceRange
        minPrice={minPrice}
        maxPrice={maxPrice}
        onMinPriceChange={setMinPrice}
        onMaxPriceChange={setMaxPrice}
      />

      {hasActiveFilters && (
        <div className="pt-4 border-t border-border">
          <ClearFiltersButton onClick={handleClearFilters} />
        </div>
      )}
    </div>
  );
}
