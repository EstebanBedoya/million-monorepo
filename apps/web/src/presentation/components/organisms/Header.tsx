'use client';

import { ThemeToggle } from '@/presentation/components/molecules/ThemeToggle';
import { LanguageSwitcher } from '@/presentation/components/molecules/LanguageSwitcher';
import { type Locale } from '@/i18n';

interface HeaderProps {
  currentLang: Locale;
}

export function Header({ currentLang }: HeaderProps) {
  return (
    <header className="sticky top-0 z-50 w-full border-b border-border bg-background/95 backdrop-blur supports-[backdrop-filter]:bg-background/60">
      <div className="container mx-auto px-4 h-16 flex items-center justify-between">
        {/* Logo/Brand */}
        <div className="flex items-center space-x-2">
          <h1 className="text-xl font-bold text-foreground">MILLION</h1>
        </div>

        {/* Controls */}
        <div className="flex items-center space-x-4">
          <LanguageSwitcher currentLang={currentLang} />
          <ThemeToggle />
        </div>
      </div>
    </header>
  );
}
