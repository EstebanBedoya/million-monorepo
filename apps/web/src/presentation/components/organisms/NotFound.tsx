'use client';

import Link from 'next/link';
import { Home, Search } from 'lucide-react';
import { Button } from '../atoms/Button';
import { Icon } from '../atoms/Icon';

export interface NotFoundProps {
  title?: string;
  message?: string;
  showSearchButton?: boolean;
}

export const NotFound = ({ 
  title = 'Property Not Found',
  message = "We couldn't find the property you're looking for. It may have been removed or is no longer available.",
}: NotFoundProps) => {
  return (
    <div className="min-h-[60vh] flex items-center justify-center px-4">
      <div className="text-center max-w-md">
        <div className="mb-8 flex justify-center">
          <div className="w-24 h-24 rounded-full bg-secondary/10 flex items-center justify-center">
            <Icon 
              icon={Search} 
              size="xl" 
              className="text-secondary"
              aria-hidden={true}
            />
          </div>
        </div>
        
        <h1 className="text-4xl font-bold text-foreground mb-4">
          404
        </h1>
        
        <h2 className="text-2xl font-semibold text-foreground mb-3">
          {title}
        </h2>
        
        <p className="text-secondary mb-8">
          {message}
        </p>
        
        <div className="flex flex-col sm:flex-row gap-3 justify-center">
          <Link href="/properties">
            <Button className="w-full sm:w-auto">
              <Icon icon={Home} size="sm" className="mr-2" />
              Back to Properties
            </Button>
          </Link>
        </div>
      </div>
    </div>
  );
};

