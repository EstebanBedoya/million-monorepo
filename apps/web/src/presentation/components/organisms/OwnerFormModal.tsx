'use client';

import { useState, useEffect } from 'react';
import { createPortal } from 'react-dom';
import { X } from 'lucide-react';
import { useDictionary } from '@/i18n/client';
import { CreateOwnerDto } from '@/domain/entities/Owner';
import { Button } from '@/presentation/components/atoms/Button';
import { Input } from '@/presentation/components/atoms/Input';
import { Label } from '@/presentation/components/atoms/Label';

interface OwnerFormModalProps {
  isOpen: boolean;
  onClose: () => void;
  onSubmit: (data: CreateOwnerDto) => Promise<void>;
}

export function OwnerFormModal({
  isOpen,
  onClose,
  onSubmit,
}: OwnerFormModalProps) {
  const dict = useDictionary();
  const [formData, setFormData] = useState<CreateOwnerDto>({
    name: '',
    address: '',
    photo: '',
    birthday: '',
  });
  
  const [isLoading, setIsLoading] = useState(false);
  const [errors, setErrors] = useState<Record<string, string>>({});
  const [mounted, setMounted] = useState(false);

  useEffect(() => {
    setMounted(true);
    return () => setMounted(false);
  }, []);

  useEffect(() => {
    if (isOpen) {
      setFormData({
        name: '',
        address: '',
        photo: '',
        birthday: '',
      });
      setErrors({});
    }
  }, [isOpen]);

  useEffect(() => {
    if (isOpen) {
      document.body.style.overflow = 'hidden';
    } else {
      document.body.style.overflow = 'unset';
    }
    return () => {
      document.body.style.overflow = 'unset';
    };
  }, [isOpen]);

  const validate = (): boolean => {
    const newErrors: Record<string, string> = {};

    if (!formData.name.trim()) {
      newErrors.name = dict.propertyForm?.required || 'This field is required';
    }
    if (!formData.address.trim()) {
      newErrors.address = dict.propertyForm?.required || 'This field is required';
    }
    if (!formData.birthday) {
      newErrors.birthday = dict.propertyForm?.required || 'This field is required';
    }

    setErrors(newErrors);
    return Object.keys(newErrors).length === 0;
  };

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();

    if (!validate()) {
      return;
    }

    setIsLoading(true);
    try {
      await onSubmit(formData);
      onClose();
    } catch (error) {
      console.error('Error creating owner:', error);
      setErrors({ submit: 'Failed to create owner' });
    } finally {
      setIsLoading(false);
    }
  };

  const handleChange = (field: keyof CreateOwnerDto, value: string) => {
    setFormData(prev => ({ ...prev, [field]: value }));
    if (errors[field]) {
      setErrors(prev => {
        const newErrors = { ...prev };
        delete newErrors[field];
        return newErrors;
      });
    }
  };

  if (!isOpen || !mounted) return null;

  const modalContent = (
    <div className="fixed inset-0 z-[9999] overflow-y-auto">
      <div className="flex min-h-screen items-center justify-center p-4">
        <div 
          className="fixed inset-0 bg-black/70 backdrop-blur-sm"
          onClick={onClose}
        />
        
        <div className="relative w-full max-w-md transform overflow-hidden rounded-lg bg-card shadow-2xl transition-all border border-border z-[10000]">
          <div className="bg-card px-6 py-4 border-b border-border">
            <div className="flex items-center justify-between">
              <h2 className="text-xl font-semibold text-foreground">
                {dict.ownerForm?.createTitle || 'Create New Owner'}
              </h2>
              <button
                onClick={onClose}
                className="text-secondary hover:text-foreground transition-colors"
              >
                <X className="w-6 h-6" />
              </button>
            </div>
          </div>

          <form onSubmit={handleSubmit} className="bg-card px-6 py-4">
            <div className="space-y-4">
              <div>
                <Label required className="mb-2">
                  {dict.ownerForm?.name || 'Name'}
                </Label>
                <Input
                  type="text"
                  value={formData.name}
                  onChange={(e) => handleChange('name', e.target.value)}
                  error={!!errors.name}
                  placeholder="John Doe"
                />
                {errors.name && <p className="mt-1 text-sm text-red-500">{errors.name}</p>}
              </div>

              <div>
                <Label required className="mb-2">
                  {dict.ownerForm?.address || 'Address'}
                </Label>
                <Input
                  type="text"
                  value={formData.address}
                  onChange={(e) => handleChange('address', e.target.value)}
                  error={!!errors.address}
                  placeholder="123 Main St, City"
                />
                {errors.address && <p className="mt-1 text-sm text-red-500">{errors.address}</p>}
              </div>

              <div>
                <Label className="mb-2">
                  {dict.ownerForm?.photo || 'Photo URL'}
                </Label>
                <Input
                  type="text"
                  value={formData.photo}
                  onChange={(e) => handleChange('photo', e.target.value)}
                  placeholder="https://..."
                />
              </div>

              <div>
                <Label required className="mb-2">
                  {dict.ownerForm?.birthday || 'Birthday'}
                </Label>
                <Input
                  type="date"
                  value={formData.birthday}
                  onChange={(e) => handleChange('birthday', e.target.value)}
                  error={!!errors.birthday}
                />
                {errors.birthday && <p className="mt-1 text-sm text-red-500">{errors.birthday}</p>}
              </div>

              {errors.submit && (
                <div className="p-3 bg-red-50 border border-red-200 rounded-md">
                  <p className="text-sm text-red-700">{errors.submit}</p>
                </div>
              )}
            </div>

            <div className="flex justify-end gap-3 mt-6 pt-4 border-t border-border">
              <Button
                type="button"
                onClick={onClose}
                disabled={isLoading}
                variant="secondary"
                size="md"
              >
                {dict.propertyForm?.cancel || 'Cancel'}
              </Button>
              <Button
                type="submit"
                disabled={isLoading}
                loading={isLoading}
                variant="primary"
                size="md"
              >
                {dict.propertyForm?.submit || 'Save'}
              </Button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );

  return createPortal(modalContent, document.body);
}

