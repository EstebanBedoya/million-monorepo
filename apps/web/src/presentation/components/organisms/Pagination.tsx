'use client';

import { PageButton } from '../molecules/PageButton';
import { PageNumberButton } from '../molecules/PageNumberButton';

interface PaginationProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
  hasNext: boolean;
  hasPrev: boolean;
}

export function Pagination({ 
  currentPage, 
  totalPages, 
  onPageChange, 
  hasNext, 
  hasPrev 
}: PaginationProps) {
  // Generate page numbers to display
  const getPageNumbers = () => {
    const pages: (number | string)[] = [];
    const maxVisible = 5; // Maximum number of page buttons to show

    if (totalPages <= maxVisible) {
      // Show all pages if total is less than max
      for (let i = 1; i <= totalPages; i++) {
        pages.push(i);
      }
    } else {
      // Always show first page
      pages.push(1);

      if (currentPage > 3) {
        pages.push('...');
      }

      // Show pages around current page
      const start = Math.max(2, currentPage - 1);
      const end = Math.min(totalPages - 1, currentPage + 1);

      for (let i = start; i <= end; i++) {
        pages.push(i);
      }

      if (currentPage < totalPages - 2) {
        pages.push('...');
      }

      // Always show last page
      pages.push(totalPages);
    }

    return pages;
  };

  if (totalPages <= 1) {
    return null;
  }

  return (
    <nav className="flex items-center justify-center gap-2 mt-8" role="navigation" aria-label="Pagination">
      {/* Previous button */}
      <PageButton
        direction="prev"
        disabled={!hasPrev}
        onClick={() => onPageChange(currentPage - 1)}
      />

      {/* Page numbers */}
      <div className="flex items-center gap-1">
        {getPageNumbers().map((page, index) => {
          if (page === '...') {
            return (
              <span 
                key={`ellipsis-${index}`} 
                className="px-3 py-2 text-secondary"
                aria-hidden="true"
              >
                ...
              </span>
            );
          }

          const pageNumber = page as number;
          const isActive = pageNumber === currentPage;

          return (
            <PageNumberButton
              key={pageNumber}
              page={pageNumber}
              isActive={isActive}
              onClick={() => onPageChange(pageNumber)}
            />
          );
        })}
      </div>

      {/* Next button */}
      <PageButton
        direction="next"
        disabled={!hasNext}
        onClick={() => onPageChange(currentPage + 1)}
      />
    </nav>
  );
}
