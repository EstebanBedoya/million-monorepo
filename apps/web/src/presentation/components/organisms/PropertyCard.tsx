'use client';

import { useRouter } from 'next/navigation';
import { Property } from '../../../domain/entities/Property';
import { Image } from '../atoms/Image';
import { Badge } from '../atoms/Badge';
import { Price } from '../atoms/Price';
import { Button } from '../atoms/Button';
import { LocationInfo } from '../molecules/LocationInfo';
import { PropertyDetails } from '../molecules/PropertyDetails';

export interface PropertyCardProps {
  property: Property;
  onViewDetails?: (property: Property) => void;
}

export function PropertyCard({ property, onViewDetails }: PropertyCardProps) {
  const router = useRouter();

  const handleViewDetails = () => {
    if (onViewDetails) {
      onViewDetails(property);
    } else {
      router.push(`/properties/${property.id}`);
    }
  };

  return (
    <article className="card-elevated overflow-hidden group cursor-pointer" role="article">
      <div className="relative h-64 overflow-hidden bg-secondary/10">
        <Image
          src={property.images[0]}
          alt={property.name}
          fill
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          className="object-cover transition-transform duration-500 group-hover:scale-110"
          fallbackSrc="/placeholder-property.jpg"
        />
        
        {property.propertyType && (
          <div className="absolute top-4 left-4">
            <Badge variant="secondary" className="bg-gray-700/50 text-gray-100 backdrop-blur-sm shadow-sm">
              {property.propertyType}
            </Badge>
          </div>
        )}
      </div>

      <div className="p-6">
        <div className="mb-3">
          <Price amount={property.price} size="xl" />
        </div>

        <h3 className="text-xl font-semibold text-foreground mb-2 line-clamp-1 group-hover:text-accent transition-colors">
          {property.name}
        </h3>

        <LocationInfo 
          address={property.location}
          className="mb-4"
        />

        <PropertyDetails
          bedrooms={property.bedrooms}
          bathrooms={property.bathrooms}
          area={property.area}
          areaUnit={property.areaUnit}
        />

        <Button
          onClick={handleViewDetails}
          className="mt-4 w-full"
          aria-label={`View details for ${property.name}`}
        >
          View Details
        </Button>
      </div>
    </article>
  );
}
