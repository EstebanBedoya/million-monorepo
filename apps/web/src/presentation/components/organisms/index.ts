export { PropertyCard } from './PropertyCard';
export { FiltersBar } from './FiltersBar';
export { Sidebar } from './Sidebar';
export { Pagination } from './Pagination';
export { NotFound } from './NotFound';
export { PropertyFormModal } from './PropertyFormModal';
export { Header } from './Header';
export { PropertyForm } from './PropertyForm';
export { OwnerSelector } from './OwnerSelector';
export { OwnerFormModal } from './OwnerFormModal';
export { UnsplashImageGallery } from './UnsplashImageGallery';
