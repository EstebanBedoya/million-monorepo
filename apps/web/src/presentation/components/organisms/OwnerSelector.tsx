'use client';

import { useState, useEffect, useMemo } from 'react';
import { Plus } from 'lucide-react';
import { useDictionary } from '@/i18n/client';
import { Owner, CreateOwnerDto } from '@/domain/entities/Owner';
import { OwnerFormModal } from './OwnerFormModal';
import { Button } from '@/presentation/components/atoms/Button';
import { Label } from '@/presentation/components/atoms/Label';
import { Autocomplete, AutocompleteOption } from '@/presentation/components/molecules/Autocomplete';
import { Container } from '@/infrastructure/di/Container';
import { GetOwnersUseCase } from '@/application/use-cases/GetOwnersUseCase';
import { CreateOwnerUseCase } from '@/application/use-cases/CreateOwnerUseCase';

interface OwnerSelectorProps {
  selectedOwnerId: string;
  onOwnerChange: (ownerId: string) => void;
  error?: string;
}

export function OwnerSelector({
  selectedOwnerId,
  onOwnerChange,
  error,
}: OwnerSelectorProps) {
  const dict = useDictionary();
  const [owners, setOwners] = useState<Owner[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [isModalOpen, setIsModalOpen] = useState(false);

  const container = Container.getInstance();
  const getOwnersUseCase = container.get<GetOwnersUseCase>('GetOwnersUseCase');
  const createOwnerUseCase = container.get<CreateOwnerUseCase>('CreateOwnerUseCase');

  useEffect(() => {
    loadOwners();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const loadOwners = async (search?: string) => {
    setIsLoading(true);
    try {
      const result = await getOwnersUseCase.execute(1, 100, search);
      setOwners(result.owners);
    } catch (error) {
      console.error('Error loading owners:', error);
    } finally {
      setIsLoading(false);
    }
  };

  const ownerOptions: AutocompleteOption[] = useMemo(() => {
    return owners.map((owner) => ({
      value: owner.idOwner,
      label: owner.name,
      description: owner.address,
    }));
  }, [owners]);

  const handleCreateOwner = async (data: CreateOwnerDto) => {
    try {
      const newOwner = await createOwnerUseCase.execute(data);
      setOwners([newOwner, ...owners]);
      onOwnerChange(newOwner.idOwner);
      setIsModalOpen(false);
    } catch (error) {
      console.error('Error creating owner:', error);
      throw error;
    }
  };

  return (
    <div>
      <Label required className="mb-2">
        {dict.propertyForm?.owner || 'Owner'}
      </Label>
      
      <div className="flex gap-2">
        <Autocomplete
          options={ownerOptions}
          value={selectedOwnerId}
          onChange={onOwnerChange}
          onSearch={loadOwners}
          placeholder={dict.propertyForm?.selectOwner || 'Search and select an owner...'}
          emptyMessage="No owners found"
          loading={isLoading}
          error={!!error}
          className="flex-1"
        />

        <Button
          type="button"
          onClick={() => setIsModalOpen(true)}
          variant="primary"
          size="md"
          className="whitespace-nowrap"
          title={dict.propertyForm?.createNewOwner || 'Create New Owner'}
        >
          <Plus className="w-5 h-5 sm:mr-2" />
          <span className="hidden sm:inline">
            {dict.propertyForm?.createNewOwner || 'New'}
          </span>
        </Button>
      </div>

      {error && <p className="mt-1 text-sm text-red-500">{error}</p>}

      <OwnerFormModal
        isOpen={isModalOpen}
        onClose={() => setIsModalOpen(false)}
        onSubmit={handleCreateOwner}
      />
    </div>
  );
}

