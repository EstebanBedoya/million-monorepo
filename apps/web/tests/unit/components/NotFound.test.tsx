import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { NotFound } from '@/presentation/components/organisms/NotFound';

// Mock useDictionary and useLocale
jest.mock('@/i18n/client', () => ({
  useDictionary: () => ({
    notFound: {
      code: '404',
      title: 'Page Not Found',
      message: 'The page you are looking for does not exist',
      backToProperties: 'Back to Properties',
    },
  }),
  useLocale: () => 'en',
}));

describe('NotFound', () => {
  it('should render default title and message from dictionary', () => {
    render(<NotFound />);
    expect(screen.getByText('404')).toBeInTheDocument();
    expect(screen.getByText('Page Not Found')).toBeInTheDocument();
    expect(screen.getByText('The page you are looking for does not exist')).toBeInTheDocument();
  });

  it('should render custom title when provided', () => {
    render(<NotFound title="Custom Title" />);
    expect(screen.getByText('Custom Title')).toBeInTheDocument();
  });

  it('should render custom message when provided', () => {
    render(<NotFound message="Custom error message" />);
    expect(screen.getByText('Custom error message')).toBeInTheDocument();
  });

  it('should render back to properties button', () => {
    render(<NotFound />);
    expect(screen.getByText('Back to Properties')).toBeInTheDocument();
  });

  it('should have link to properties page with correct locale', () => {
    const { container } = render(<NotFound />);
    const link = container.querySelector('a[href="/en/properties"]');
    expect(link).toBeInTheDocument();
  });

  it('should render both custom title and message', () => {
    render(<NotFound title="Oops!" message="Something went wrong" />);
    expect(screen.getByText('Oops!')).toBeInTheDocument();
    expect(screen.getByText('Something went wrong')).toBeInTheDocument();
  });
});

