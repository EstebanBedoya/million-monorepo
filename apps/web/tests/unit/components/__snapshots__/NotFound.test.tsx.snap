// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should match snapshot 1`] = `
<div
  class="min-h-[60vh] flex items-center justify-center px-4"
>
  <div
    class="text-center max-w-md"
  >
    <div
      class="mb-8 flex justify-center"
    >
      <div
        class="w-24 h-24 rounded-full bg-secondary/10 flex items-center justify-center"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-search w-8 h-8 text-secondary"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m21 21-4.34-4.34"
          />
          <circle
            cx="11"
            cy="11"
            r="8"
          />
        </svg>
      </div>
    </div>
    <h1
      class="text-4xl font-bold text-foreground mb-4"
    >
      404
    </h1>
    <h2
      class="text-2xl font-semibold text-foreground mb-3"
    >
      Property Not Found
    </h2>
    <p
      class="text-secondary mb-8"
    >
      We couldn't find the property you're looking for. It may have been removed or is no longer available.
    </p>
    <div
      class="flex flex-col sm:flex-row gap-3 justify-center"
    >
      <a
        href="/properties"
      >
        <button
          class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed bg-accent text-white hover:bg-accent/90 focus:ring-accent h-10 px-4 text-sm w-full sm:w-auto"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-house w-4 h-4 text-current mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
            />
            <path
              d="M3 10a2 2 0 0 1 .709-1.528l7-6a2 2 0 0 1 2.582 0l7 6A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
            />
          </svg>
          Back to Properties
        </button>
      </a>
    </div>
  </div>
</div>
`;
